sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -4133,7 +4133,7 @@ pi_result cuda_piextUSMHostAlloc(void **result_ptr, pi_context context,
   } catch (pi_result error) {
     result = error;
   }
-  assert(reinterpret_cast<std::uintptr_t>(*result_ptr) % alignment == 0);
```

Comment:
I see the issue when `alignment == 0` - but removing it there are no checks that the allocation respects the alignment ?

Reply:
@fwyzard, do you suggest adding something like:
```c++
assert(alignment && (reinterpret_cast<std::uintptr_t>(*result_ptr) % alignment == 0));
```

Reply:
I was actually thinking
```c++
assert(alignment == 0 || reinterpret_cast<std::uintptr_t>(*result_ptr) % alignment == 0);
```
since `alignment == 0` means no alignment requirements IIRC ?

Reply:
Right. This seems to be an easy change: the tests are already enabled, so it should be just three new lines of code in `pi_cuda.cpp`.

Reply:
https://github.com/intel/llvm/pull/2831 recovers assertions.